atest/testdata/misc/parent_suite/child_suite2/test.robot

```diff
@@ -0,0 +1,7 @@
+*** Settings ***
+Documentation  Should not execute
+Documentation  Should not execute
+
+*** Test Cases ***
+Child Suite Test
+  Log  Should not execute!
```

Comment:
This test suite, as well as the other added suite, is executed by other tests running `atest/testdata/misc`. There are some tests that validate that correct suites/tests are run in that case, and they are going to fail because new suites/tests are added. Also, it's a bit confusing that a suite with documentation `Should not execute` (twice) is actually executed. If these suites are added, the affected tests must naturally be fixed.

An alternative would be just using the existing stuff under `atest/testdata/misc`. It seems that the main reason this suite is added is to validate it is not parsed at all, and that is done by looking there's no error about duplicate documentation in the stderr. Could the actual test be changed to use the existing `warnings_and_errors.robot` somehow? It does't have duplicate documentation, but it has a non-existing setting that causes a similar error.

Reply:
Went on and used the first proposal in the updated PR.
I completly re-done the directory structure, now we will have two suites:
-	Suite Should.Be.Parsed
-	Suite Should.Not Be.Parsed

The second suite contains a test with a non-existing setting that should not be normally encountered when executing with command line option `--suite suite_should.be.parsed`. I also updated the unit tests with the new suites/tests.

Regarding the alternative proposal, I thought about different ways to do this but didn't find one that could rely only on the `warnings_and_errors.robot`. Or to use it elegantly with another test file. If you could give me a suggestion I would glady update the PR.